packages/go/cypher/model/pgsql/slices.go

```diff
@@ -0,0 +1,34 @@
+package pgsql
+
+import "fmt"
+
+func SliceAs[T any, TS []T, F any, FS []F](fs FS) (TS, error) {
+	ts := make(TS, len(fs))
+
+	for idx := 0; idx < len(fs); idx++ {
+		if tTyped, isTType := any(fs[idx]).(T); !isTType {
+			var emptyT T
+			return nil, fmt.Errorf("slice type %T does not convert to %T", fs[idx], emptyT)
+		} else {
+			ts[idx] = tTyped
+		}
+	}
+
+	return ts, nil
+}
```

Comment:
We could probably simplify wherever this is being used by instead using `slicesext.Map[F, T any](fn func(F) T, fs []F) []T` where `fn` is a transformation function that leverages the compiler for type coercion.

Reply:
Implemented with some minor tweaks